src/topology/metric_space/basic.lean

```diff
@@ -672,6 +672,25 @@ lemma is_open_singleton_iff {X : Type*} [metric_space X] {x : X} :
   is_open ({x} : set X) ↔ ∃ ε > 0, ∀ y, dist y x < ε → y = x :=
 by simp [is_open_iff, subset_singleton_iff, mem_ball]
 
+/-- Given a point `x` in a discrete subset `s` of a metric space, there is an open ball
+centered at `x` and intersecting `s` only at `x`. -/
+lemma ball_of_mem_discrete [discrete_topology s] {x : α} (hx : x ∈ s) :
+∃ ε > 0, ∀ y ∈ metric.ball x ε, y ≠ x → y ∉ s :=
+begin
+  rcases disjoint_nhds_within_of_mem_discrete hx with ⟨U, U_in, hU⟩,
+  rcases metric.nhds_within_basis_ball.mem_iff.mp U_in with ⟨ε, ε_pos, hε⟩,
+  refine ⟨ε, ε_pos, λ y hyB hxy hyt, _⟩,
+  exact eq_empty_iff_forall_not_mem.mp (disjoint_iff_inter_eq_empty.mp hU) y ⟨hε ⟨hyB, hxy⟩, hyt⟩,
+end
+
+/-- Given a point `x` in a discrete subset `s` of a metric space, there is a radius `ε` such that
+`x` is the only point of `s` with distance smaller than `ε` from `x`.
+This formulation is defeq to `metric.ball_of_mem_discrete`.
+ -/
+lemma dist_lt_of_mem_discrete [discrete_topology s] {x : α} (hx : x ∈ s) :
+∃ ε > 0, ∀ y, dist y x < ε → y ≠ x → y ∉ s :=
+ball_of_mem_discrete hx
```

Comment:
I'm not sure we need this version, I wrote it to point out what the preceding version unfolds to, but this was only for your convenience when using it.  I don't really know what is the policy here. We could argue we could keep this lemma, use it, and if the definition of `metric.ball` ever change then we will only have to fix this proof.

Reply:
I see.  I am still confused by when defeq matters and when it does not.  This is why I thought that I should include this lemma in this case!  I have no specific attachment to it: I used this version in the Liouville argument, but I will switch to the other version instead.

Reply:
I removed the defeq version.